plugins/org.eclipse.elk.core.service/src/org/eclipse/elk/core/service/DiagramLayoutEngine.java

```diff
@@ -193,19 +150,51 @@ public LayoutConfigurator addLayoutRun() {
         }
     }
     
-    private final LayoutConfigurationManager configManager = new LayoutConfigurationManager();
-    
-    /** the graph layout engine for executing layout algorithms on the hierarchy levels of a graph. */
-    private final IGraphLayoutEngine graphLayoutEngine = new RecursiveGraphLayoutEngine((String input) -> {
-        return configManager.getAlgorithm(input);
-    });
+    /** preference identifier for debug graph output. */
+    public static final String PREF_DEBUG_OUTPUT = "elk.debug.graph";
+    /** preference identifier for execution time measurement. */
+    public static final String PREF_EXEC_TIME_MEASUREMENT = "elk.exectime.measure";
```

Comment:
These are controlled by the preference page, not by the layout configuration. We could make them support both by introducing a special property that queries the preference store when asked for a default value.